pkg/asset/agent/image/baseiso.go

```diff
@@ -91,6 +105,29 @@ func (i *BaseIso) Dependencies() []asset.Asset {
 	}
 }
 
+func (i *BaseIso) checkReleasePayloadBaseISOVersion(r Release, archName string) error {
+	logrus.Debugf("Checking release payload base ISO version")
+
+	// Get current release payload CoreOS version
+	payloadRelease, err := r.GetBaseIsoVersion(archName)
+	if err != nil {
+		return err
+	}
+
+	// Get pinned version from installer
+	metal, err := getMetalArtifact(archName)
+	if err != nil {
+		return err
+	}
+
+	// Check for a mismatch
+	if metal.Release != payloadRelease {
+		logrus.Warnf("base ISO version mismatch in release payload. Expected version %s but found %s", metal.Release, payloadRelease)
```

Comment:
If we wanted to get more sophisticated, we could try falling back to downloading from the URL if the coreos-stream in the release payload is out of date.
Error handling will get complicated though, because if we are in a disconnected environment, _that_ will fail and we'll want to carry on with the ISO from the release payload + a warning.